src/dodal/p45.py

```diff
@@ -0,0 +1,73 @@
+from ophyd import Component as Cpt
+from ophyd import EpicsMotor, MotorBundle
+from ophyd.areadetector.base import ADComponent as Cpt
+from pydantic import BaseSettings
+
+from dodal.devices.areadetector import AdAravisDetector
+
+
+# Settings can be customized via environment variables
+class Settings(BaseSettings):
+    pv_prefix: str = "BL45P"
```

Comment:
Nit: To override this we would need to have the `PV_PREFIX` environment variable set, right? I was thinking we could instead look at the `BEAMLINE` environment variable, which we already have on beamlines, and create the prefix based on that or use a simulator if it doesn't exist. e.g. like https://github.com/DiamondLightSource/python-artemis/pull/522/files#diff-2ee96bef30cad81682a38f7f9c30ffb387bc82f29e5f09393b7a3b5f2c88af7e. This would mean it'll work seamlessly on beamlines and dev machines as they are currently

Reply:
I did think about that, my main issue was that `$BEAMLINE` takes the form `Ixx` where the prefix is `BLxxI`. As you say, we could write some logic to convert between the two but then we bake in the assumption of those naming conventions. @coretl, how do controls feel, having lived with those conventions for 20 years? Is is something we'd be happy to keep going into the D-II era or should we be thinking about a more flexible approach?

Reply:
There are some annoying inconsistencies about converting from `Ixx` to `BLxxI` especially for some branchlines, so I think the pv_prefix is safer...

Reply:
I don't really like that this means for a developer I need to make sure to have `PV_PREFIX=S03` in my environment, then if I want to switch working to i04 I would need to switch to `PV_PREFIX=S04` before starting. It seems we need an environment variable that defines if you are on a live machine and reverts to the sim PVs if not. We could add `PV_PREFIX` to the beamline machines?

Reply:
Hmm, my first thought was that the adsim case (hostname based) and the s03 case (static prefix based) are different and require different solutions.

```python
# adsim
class Settings(BaseSettings):
    pv_prefix: str = hostname

_settings = Settings()

# i03
class Settings(BaseSettings):
    simulation: bool = False

_settings = Settings()
_pv_prefix = "BL03S" if _settings.simulation else "BL03I"
```

However, I also remember we've previously discussed the possibility of prefixing the S03 PVs with hostnames or other differentiators so that multiple instances can be run on the same network, do you have any more thoughts in that direction?

Reply:
> There are some annoying inconsistencies about converting from `Ixx` to `BLxxI` especially for some branchlines, so I think the pv_prefix is safer...

Although this might be a sensible thing to do if we can override the PV prefixes with some other mechanism